src/VectorizeLoops.cpp

```diff
@@ -1015,19 +1087,30 @@ class VectorSubs : public IRMutator {
 
         // Rewrite loads and stores to this allocation like so:
         // foo[x] -> foo[x*lanes + v]
-        string v = unique_name('v');
-        body = RewriteAccessToVectorAlloc(v, op->name, lanes).mutate(body);
+        vector<string> vs;
+        for (const auto &vv : vectorized_vars) {
+            string v = unique_name('v');
+            body = RewriteAccessToVectorAlloc(v, op->name, vv.lanes).mutate(body);
+            vs.push_back(v);
+            scope.push(v, Variable::make(Int(32), vv.name + ".from_zero"));
+        }
 
-        scope.push(v, Ramp::make(0, 1, lanes));
         body = mutate(body);
-        scope.pop(v);
+        for (const auto &v : vs) {
+            scope.pop(v);
+        }
 
         // Replace the widened 'v' with the actual ramp
-        // foo[x*lanes + widened_v] -> foo[x*lanes + ramp(0, 1, lanes)]
-        body = substitute(v + widening_suffix, Ramp::make(0, 1, lanes), body);
+        // foo[x*lanes + widened_v] -> foo[x*lanes + ramp(...)]
+        for (size_t ix = 0; ix < vectorized_vars.size(); ix++) {
```

Comment:
Why do the widened vars get substituted in at this point? Haven't references to them already been replaced when visiting them?

Reply:
What is replaced here is not one of the vars from Let/LetStmt, but a variable which was created in this function to sort of "vectorize" allocation. They are not strictly "widened", but they also are added to scope and such, and overall are quite similar to true widened vars, so I reused infrastructure and kept them in the same map.

Reply:
Update: this is handled differently now, vars which were are added inside of the visit(Allocate* ) node are just directly replaced with proper combination of ramps/broadcasts whenever we encounter them in visit(Var* ).